File: src/net/minecraft/client/render/entity/GhastEntityRenderer.cpp
#include "net/minecraft/client/render/entity/EntityRenderers.hpp"
#include "net/minecraft/client/render/entity/model/GhastEntityModel.hpp"
#include "net/minecraft/entity/mob/GhastEntity.hpp"
#include "net/minecraft/client/render/RenderSystem.hpp"
namespace net::minecraft::client::render::entity {
GhastEntityRenderer::GhastEntityRenderer() : LivingEntityRenderer(new model::GhastEntityModel(), 0.5f) {
}
void GhastEntityRenderer::applyScale(const net::minecraft::LivingEntity& entity, float tickDelta) {
 const auto* ghast = dynamic_cast<const ::net::minecraft::entity::mob::GhastEntity*>(&entity);
 if(ghast == nullptr) {
  return;
 }
 float charge = (static_cast<float>(ghast->lastChargeTime) +
                 static_cast<float>(ghast->chargeTime - ghast->lastChargeTime) * tickDelta) /
                20.0f;
 if(charge < 0.0f) {
  charge = 0.0f;
 }
 charge = 1.0f / (charge * charge * charge * charge * charge * 2.0f + 1.0f);
 const float scaleY = (8.0f + charge) / 2.0f;
 const float scaleXZ = (8.0f + 1.0f / charge) / 2.0f;
 RenderSystem::scale(scaleXZ, scaleY, scaleXZ);
 RenderSystem::color4f(1.0f, 1.0f, 1.0f, 1.0f);
}
} // namespace net::minecraft::client::render::entity
#include "net/minecraft/client/entity/EntityClientRendererRegistration.hpp"
#include "net/minecraft/entity/mob/GhastEntity.hpp"
namespace net::minecraft::entity::mob {
std::unique_ptr<::net::minecraft::client::render::entity::EntityRenderer> GhastEntity::ClientRenderer::create() {
 return std::make_unique<::net::minecraft::client::render::entity::GhastEntityRenderer>();
}
} // namespace net::minecraft::entity::mob
namespace {
static ::net::minecraft::registry::RegisterEntityRenderer<net::minecraft::entity::mob::GhastEntity> autoRendererReg;
} // namespace

File: mcp/src/net/minecraft/client/render/entity/GhastEntityRenderer.java
/*
 * Decompiled with CFR 0.1.1 (FabricMC 57d88659).
 * 
 * Could not load the following classes:
 *  net.fabricmc.api.EnvType
 *  net.fabricmc.api.Environment
 */
package net.minecraft.client.render.entity;

import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.render.entity.LivingEntityRenderer;
import net.minecraft.client.render.entity.model.GhastEntityModel;
import net.minecraft.entity.mob.GhastEntity;
import org.lwjgl.opengl.GL11;

@Environment(value=EnvType.CLIENT)
public class GhastEntityRenderer
extends LivingEntityRenderer {
    public GhastEntityRenderer() {
        super(new GhastEntityModel(), 0.5f);
    }

    protected void applyScale(GhastEntity arg, float f) {
        GhastEntity ghastEntity = arg;
        float f2 = ((float)ghastEntity.lastChargeTime + (float)(ghastEntity.chargeTime - ghastEntity.lastChargeTime) * f) / 20.0f;
        if (f2 < 0.0f) {
            f2 = 0.0f;
        }
        f2 = 1.0f / (f2 * f2 * f2 * f2 * f2 * 2.0f + 1.0f);
        float f3 = (8.0f + f2) / 2.0f;
        float f4 = (8.0f + 1.0f / f2) / 2.0f;
        GL11.glScalef(f4, f3, f4);
        GL11.glColor4f(1.0f, 1.0f, 1.0f, 1.0f);
    }
}

